### PR TITLE
Add MixGo CE board

### PR DESCRIPTION
**Pull request by dahanzimin**
Thank circuitpython for adding a new board to circuitpython ， I hope it can pass

### CODE REVIEW

File: ports/espressif/boards/mixgo_ce_udisk/mpconfigboard.mk [hunk 1]
```diff
@@ -0,0 +1,24 @@
+USB_VID = 0x239A
+USB_PID = 0x80A8
```
Comment (tannewt) [inline ports/espressif/boards/mixgo_ce_udisk/mpconfigboard.mk:2]: You'll need your own USB VID/PID combination. Espressif usually gives them out here: https://github.com/espressif/usb-pids

**Review by tannewt — DISMISSED** [review]
Why are you including the library as a frozen?

**Comment by dhalbert** [comment]
@dahanzimin Are you still awaiting the USB PID from Espressif, or is the one in the commit what you got?

**Comment by dahanzimin** [comment]
@dhalbert ,Thank you for your attention. I have applied for PID in Lexin ands ubmitted them [PID](https://github.com/adafruit/circuitpython/pull/6383/commits/d4b02e22ae7fc7aa47e6ae46f9bd364e196ca57a)，I don't know what else to do next, just waiting for the merger?

> @大汉子民你还在等待来自Espressif的USB PID，还是在提交中的那个是你得到的？

**Review by dhalbert — CHANGES_REQUESTED** [review]
Thanks for the update on the USB PID.

Two comments about your custom library:

1) The usual way to do this is to keep the library in a separate repo, and add it as a submodule. This will allow you to maintain the library separately, and not depend on the `circuitpython` repo for source control. Also it reduces the size of `circuitpython` and minimizes the custom code in it.

2) Consider whether you need to freeze the library. The reason we freeze libraries into certain firmware images is only to save RAM space when importing the libraries. It is not purposely to make using the board more convenient, though that is an accidental effect. Circuit Playground Express (SAMD21, with only 32kB of RAM) has several frozen libraries for the RAM space reason. But Circuit Playground Bluefruit (nrf52840, 256kB RAM) has none. ESP32-S2 WROOM has 320kB RAM, so you are not under extremem RAM size pressure.

The disadvantage of freezing libraries is that updates require building new firmware. If you keep the libraries separate, then your users will not have to upgrade firmware merely to update the frozen libraries. Of course, they can always override the frozen libraries by placing new versions before `.frozen` in `sys.path`.

Given the size of your custom library, it is probably going to have bug fixes and improvements going forward.

**Comment by dhalbert** [comment]
Note also the pre-commit failures here: https://github.com/adafruit/circuitpython/runs/6398538029?check_suite_focus=true. For some files, there is simply extraneous whitespace or a missing newline at the end of the file. For the Python files, they may be failing `black` or `pylint` in some way. If you move to your own repo and a submodule, they will not be checked (you could of course check them yourself).

**Comment by dahanzimin** [comment]
Thank you very much for your help. We use this board to promote the programming education of circuit Python in China, and use the graphical programming software [mixly](https://mixly.org) in primary and middle school students. They often don't want complex lib operation

**Comment by dahanzimin** [comment]
> Note also the pre-commit failures here: https://github.com/adafruit/circuitpython/runs/6398538029?check_suite_focus=true. For some files, there is simply extraneous whitespace or a missing newline at the end of the file. For the Python files, they may be failing `black` or `pylint` in some way. If you move to your own repo and a submodule, they will not be checked (you could of course check them yourself).

Sorry, I will pay attention to this detail later. Thank you for your answers and help

**Comment by dhalbert** [comment]
Here is some info about using `pre-commit` to check the files: https://learn.adafruit.com/building-circuitpython/build-circuitpython#install-pre-commit-3096511-10

**Comment by dahanzimin** [comment]
@dhalbert  Thank you for your help. Now all checks have passed. What else do I need to do for this merger?

**Review by dhalbert — COMMENTED** [review]
This looks good; thank you for making the submodule.

One question before merging: why is the `sdkconfig` empty? See the others in `boards/*/sdkconfig`. Some just specify `CONFIG_LWIP_LOCAL_HOSTNAME`; others have more settings.

**Comment by dhalbert** [comment]
@dahanzimin is it ready for review again?

**Comment by dahanzimin** [comment]
> @dahanzimin is it ready for review again?

OK, thank you again. As always, I will promote circuitpython in another region. The board program has been added and can be reviewed again. Thank you

**Review by dhalbert — APPROVED** [review]
Looks good to me! Thanks for addressing our requests. I see that https://github.com/adafruit/circuitpython-org/pull/946 for the Mixgo CE board. But are there now two boards, or are they different modes of the same board? Since there are two builds, I think you will want another entry in circuitpython.org.

**Comment by dahanzimin** [comment]
Yes, the same board has different modes (considering that many computers cannot read the drive letter when programming), in circuit python org. Need two web pages to support it? Submit another (mixgo_ ce_ serial) page later.

**Comment by dhalbert** [comment]
Yes, if there are two builds, you will need another page for the other download.

**Comment by dhalbert** [comment]
@dahanzimin - the merge documentation build failed, I believe because the frozen library was in the board directory rather than in  `frozen/`. Could you submit another PR that moves the submodule to the `frozen/` directory, and change your `mpconfigboard.mk` files appropriately?

**Comment by dhalbert** [comment]
Error here: https://github.com/adafruit/circuitpython/runs/6543811274?check_suite_focus=true

**Comment by dhalbert** [comment]
To test this before pushing the new PR, go to the top level and do `make html`. You will need to do `pip3 install -r requirements.doc.txt` to get all the Python libraries to build the documentaiton.

**Comment by dhalbert** [comment]
@dahanzimin I am fixing this in #6420.

**Comment by dahanzimin** [comment]
Thank you. I'm already repairing it [dahanzimin/circuitpython/commit/](https://github.com/dahanzimin/circuitpython/commit/6454c5895b9f665d7f7ddf60155e7a6e2fe41660),[Actions](https://github.com/dahanzimin/circuitpython/actions/runs/2368760801) are in progress

**Pull request merged**